tests/immu/test_session.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2022 CodeNotary, Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from immudb import constants
+from tests.immuTestClient import ImmuTestClient
+import time
+
+
+class TestSessionTransaction:
+
+    def test_simple_unmanaged_session(self, wrappedClient: ImmuTestClient):
+        if(not wrappedClient.serverHigherOrEqualsToVersion("1.2.0")):
+            pytest.skip("Version too low")
+        txInterface = wrappedClient.client.openSession("immudb", "immudb", b"defaultdb")
+        newTx = txInterface.newTx()
+        table = wrappedClient._generateTableName()
+        newTx.sqlExec(f"CREATE TABLE {table} (id INTEGER AUTO_INCREMENT, tester VARCHAR[10], PRIMARY KEY id)")
+        commit = newTx.commit()
+        assert commit.header.id != None
+
+        newTx = txInterface.newTx()
+        newTx.sqlExec(f"INSERT INTO {table} (tester) VALUES(@testParam)", params = {"testParam": "123"})
+        what = newTx.sqlQuery(f"SELECT * FROM {table}", dict(), columnNameMode=constants.COLUMN_NAME_MODE_FIELD)
+        assert what == [{"id": 1, "tester": '123'}]
+        commit = newTx.commit()
+        assert commit.header.id != None
+
+        newTx = txInterface.newTx()
+        newTx.sqlExec(f"INSERT INTO {table} (tester) VALUES(@testParam)", params = {"testParam": "321"})
+        what = newTx.sqlQuery(f"SELECT * FROM {table}", dict(), columnNameMode=constants.COLUMN_NAME_MODE_FIELD)
+        assert what == [{"id": 1, "tester": '123'}, {"id": 2, "tester": '321'}]
+        commit = newTx.rollback()
+
+        newTx = txInterface.newTx()
+        what = newTx.sqlQuery(f"SELECT * FROM {table}", dict(), columnNameMode=constants.COLUMN_NAME_MODE_FIELD)
+        assert what == [{"id": 1, "tester": '123'}]
+        commit = newTx.commit()
+        wrappedClient.closeSession()
```

Comment:
This should be in some `finally` block - otherwise server will keep internal session state until the timeout occurs

Reply:
Done in https://github.com/codenotary/immudb-py/pull/46/commits/b53e7a2d899e34be511b255bc4fe0d6199864c61